akka-sse/src/main/scala/de/heikoseeberger/akkasse/client/EventStreamClient.scala

```diff
@@ -105,17 +105,19 @@ object EventStreamClient {
   )(implicit ec: ExecutionContext, mat: Materializer): Source[A, NotUsed] = {
 
     def getAndHandleEvents = {
-      def getAndHandle(lastEventId: Option[String]) = {
+      def getAndHandle(lastEventId: Option[String])
+        : (Future[Option[ServerSentEvent]], A) = {
         import EventStreamUnmarshalling._
         val request = {
           val r = Get(uri).addHeader(Accept(`text/event-stream`))
           lastEventId.foldLeft(r)((r, i) => r.addHeader(`Last-Event-ID`(i)))
         }
-        val events = send(request).flatMap(Unmarshal(_).to[EventStream])
+        val events      = send(request).flatMap(Unmarshal(_).to[EventStream])
+        val emptySource = Future.successful(Source.empty[ServerSentEvent])
         Source
-          .fromFuture(events)
+          .fromFuture(events.fallbackTo(emptySource))
```

Comment:
Yes indeed, there's also the option of using `.recoverWithRetries(1, { case _ => Source.empty[ServetSentEvent] })` (I tried that first, but the fallbackTo was cheaper)

(I don't think the LastElement combinator is unique enough to warrant its own GraphStage implementation, nor generic enough to be reused. :)=